compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -696,6 +706,31 @@ private String getPlatformLibPath(String groupId, String artifactId, String vers
         }
     }
 
+    /**
+     *
+     * @param platform java platform of the dependency
+     * @param groupId group id
+     * @param artifactId artifact id
+     * @param version version
+     * @return platform lib path provided by user
+     */
+    private String getPlatformLibPathFromProvided(String platform, String groupId, String artifactId, String version) {
+
+        for (Map<String, Object> platformDep :
+                this.packageContext().packageManifest().platform(platform).dependencies()) {
+            String depArtifactId = (String) platformDep.get(JarLibrary.KEY_ARTIFACT_ID);
+            String depVersion = (String) platformDep.get(JarLibrary.KEY_VERSION);
+            String depGroupId = (String) platformDep.get(JarLibrary.KEY_GROUP_ID);
+            String depFilepath = (String) platformDep.get(JarLibrary.KEY_PATH);
+            if (artifactId.equals(depArtifactId) && groupId.equals(depGroupId)
+                    && version.equals(depVersion) && depFilepath != null && !depFilepath.isEmpty()) {
```

Comment:
```suggestion
                    && version.equals(depVersion) && !"".equals(depFilepath)) {
```

Reply:
This would not work when the value is null